force-app/main/default/classes/ODRIntegrationTest.cls

```diff
@@ -335,4 +335,38 @@ public class ODRIntegrationTest {
 
         System.assertEquals(new Map<String, Integration_Log__c>(), m1);
     }
+    
+    @isTest
+    public Static void fetchPrescriptionHistoryWithSearchKey_Test(){
+        Drug__c drug = TestFactory.newDrug('Drug 1');
+        insert drug;
+        
+        Account patient = TestFactory.newPatient('Patient 1');
+        Account provider = TestFactory.newProvider('Provider 1');
+        insert new Account[] { patient, provider };
+            
+        Case saCase = TestFactory.newCase(drug, patient, provider, 'Test case 1');
+        insert saCase;
+        
+        Test.setMock(HttpCalloutMock.class, new RestMock());
```

Comment:
Also the HTTP Mock covers success path by only checking HTTP status 200, what happens if there is 400/404/500 status codes?

Reply:
Added the Negative test scenario

Reply:
I don't see them, please review and submit PR again.